ui/app/src/components/PreviewDialog/PreviewDialogContainer.tsx

```diff
@@ -26,23 +26,37 @@ import DialogFooter from '../DialogFooter';
 import SecondaryButton from '../SecondaryButton';
 import { FormattedMessage } from 'react-intl';
 import PrimaryButton from '../PrimaryButton';
-import useDetailedItem from '../../hooks/useDetailedItem';
 import { DialogBody } from '../DialogBody';
 import { useDispatch } from 'react-redux';
 import { closePreviewDialog, showCodeEditorDialog } from '../../state/actions/dialogs';
 import { batchActions } from '../../state/actions/misc';
 import { hasEditAction, isBlobUrl } from '../../utils/content';
 import { useSelection } from '../../hooks/useSelection';
+import useItemsByPath from '../../hooks/useItemsByPath';
+import { fetchSandboxItem } from '../../services/content';
+import { DetailedItem, SandboxItem } from '../../models';
+import useActiveSiteId from '../../hooks/useActiveSiteId';
 
 export function PreviewDialogContainer(props: PreviewDialogContainerProps) {
-  const { title, content, mode, url, onClose, type, mimeType, backgroundModeIndex } = props;
+  const { title, content, mode, url, path, onClose, type, mimeType, backgroundModeIndex } = props;
   const { classes, cx } = useStyles();
-  const item = useDetailedItem(url);
+  const siteId = useActiveSiteId();
+  const items = useItemsByPath();
+  const [item, setItem] = useState<DetailedItem | SandboxItem>();
   const dispatch = useDispatch();
   const guestBase = useSelection<string>((state) => state.env.guestBase);
-
   const [isLoading, setIsLoading] = useState(true);
 
+  useEffect(() => {
+    if (type === 'editor' && path) {
+      if (items[path]) {
+        setItem(items[path]);
+      } else {
+        fetchSandboxItem(siteId, path).subscribe((item) => setItem(item));
+      }
+    }
```

Comment:
Can we dispatch the fetch, retrieve from state and remove this additional local state?